tests/substrate.rs

```diff
@@ -711,6 +711,25 @@ impl Runtime {
 
         Ok(())
     }
+
+    /// Mocks a chain extension with ID 123 that does just echo back the input.
+    #[seal(0)]
+    fn call_chain_extension(
+        id: u32,
+        input_ptr: u32,
+        input_len: u32,
+        output_ptr: u32,
+        output_len_ptr: u32,
+    ) -> Result<u32, Trap> {
+        assert_eq!(id, 123, "unkown chain extension");
+        assert_eq!(read_len(mem, output_len_ptr), 16384);
+
+        let data = read_buf(mem, input_ptr, input_len);
```

Comment:
Why not, I changed it so it reverses the input data and writes that back.

Reply:
But just to mention, because we memset 0 the output buffer before calling the chain extension, this shouldn't really matter, because the input and output buffer will only be identical if the chain extension actually did copy it.